zip-0316.rst

```diff
@@ -416,8 +471,60 @@ Requirements for both Unified Addresses and Unified Viewing Keys
   that cannot be interpreted as specified above.
 
 
-Address hardening
------------------
+Adding new types
+----------------
+
+It is intended that new Receiver Types and Viewing Key Types SHOULD
+be introduced either by a modification to this ZIP or by a new ZIP,
+in accordance with the ZIP Process [#zip-0000]_.
+
+For experimentation prior to proposing a ZIP, experimental types MAY
+be added using the reserved Typecodes :math:`\mathtt{0xFFFA}` to
+:math:`\mathtt{0xFFFF}` inclusive. This provides for six simultaneous
+experiments, which can be referred to as experiments A to F. This
+should be sufficient because experiments are expected to be reasonably
+short-term, and should otherwise be either standardized in a ZIP (and
+allocated a Typecode outside this reserved range) or discontinued.
+
+
+Deriving a UIVK from a UFVK
+---------------------------
+
+Each UIVK Encoding can straightforwardly be derived from the
+corresponding UFVK Encoding, without changing the Typecode. In the
+case of a Transparent P2PKH UFVK Encoding, the UIVK Encoding is the
+same.
+
+
+Deriving a Unified Address from a UIVK
+--------------------------------------
+
+To derive a Unified Address from a UIVK we need to choose a diversifier
+index, which MUST be valid for all of the Viewing Key Types in the
+UIVK. That is,
+
+* A Sapling diversifier index MUST generate a valid diversifier as
+  defined in ZIP 32 section “Sapling diversifier derivation”
+  [#zip-0032-sapling-diversifier-derivation]_.
+* A Transparent diversifier index MUST be in the range :math:`0` to
```

Comment:
nb: should this refer specifically to a P2PKH address, or is "transparent" sufficient in conjunction with the exclusion around P2SH addresses for UIVKs?

Reply:
I think this is sufficient.